src/pocketmine/Player.php

```diff
@@ -4337,4 +4337,14 @@ public static function getChunkCacheFromData($chunkX, $chunkZ, $payload, $orderi
 		return $pk;
 	}
 
-}
+	/**
+	 * @param Effect
+	*/
+	public function addEffect(Effect $effect){//Overwrite
+		if($effect->isBad() && $this->isCreative()){
+			return false;
```

Comment:
does this method have a return value? if so, you haven't returned anything below.